device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -419,7 +419,60 @@ public void open() throws TransportException
 
         this.taskScheduler = Executors.newScheduledThreadPool(1);
 
-        openConnection();
+        if (withRetry)
+        {
+            int connectionAttempt = 0;
+            long startTime = System.currentTimeMillis();
+
+            // this loop either ends in throwing an exception when retry expires, or by a break statement upon a successful openConnection() call
+            while (true)
+            {
+                RetryPolicy retryPolicy;
+                if (isMultiplexing)
+                {
+                    retryPolicy = multiplexingRetryPolicy;
+                }
+                else
+                {
+                    retryPolicy = this.getDefaultConfig().getRetryPolicy();
+                }
+
+                try
+                {
+                    openConnection();
+                    break; // openConnection() only returns without throwing if the connection attempt was successful
+                }
+                catch (TransportException transportException)
+                {
+                    log.debug("Encountered an exception while opening the client. Consulting configured the retry policy to see if another attempt should be made.", transportException);
```

Comment:
I don't understand this. Perhaps we mean, "consulting the configured retry policy"?

Reply:
whoops, yes, this is backwards